src/store/settings/settings-reducers.ts

```diff
@@ -185,4 +185,13 @@ export const settingsReducers = createReducer<SettingsState>(settingsInitialStat
 
     return state;
   });
+
+  builder.addCase(addMarigoldRpc, state => {
+    if (!state.marigoldWasAdded && !state.rpcList.some(rpc => rpc.url === MARIGOLD_RPC.url)) {
```

Comment:
i think checking rpcList is enough here, adding new flag is unnecessary

Reply:
Sorry, it isn't because a user may remove Marigold RPC intentionally and we don't want to make it come back.